core/src/main/java/com/datastrato/gravitino/Configs.java

```diff
@@ -33,6 +34,24 @@ public interface Configs {
   Long DEFAULT_KV_DELETE_AFTER_TIME = 604800000L; // 7 days
   String KV_DELETE_AFTER_TIME_KEY = "gravitino.entity.store.kv.deleteAfterTimeMs";
 
+  // Config for data keep time after soft deletion, in milliseconds.
+  String STORE_DELETE_AFTER_TIME_KEY = "gravitino.entity.store.deleteAfterTimeMs";
+  // using the fallback default value
+  Long DEFAULT_STORE_DELETE_AFTER_TIME = DEFAULT_KV_DELETE_AFTER_TIME;
+  // The maximum allowed keep time for data deletion, in milliseconds. Equivalent to 30 days.
+  Long MAX_DELETE_TIME_ALLOW = 1000 * 60 * 60 * 24 * 30L;
+  // The minimum allowed keep time for data deletion, in milliseconds. Equivalent to 10 minutes.
+  Long MIN_DELETE_TIME_ALLOW = 1000 * 60 * 10L;
+
+  // Count of versions allowed to be retained, including the current version, used to delete old
+  // versions data.
+  String VERSION_RETENTION_COUNT_KEY = "gravitino.entity.store.versionRetentionCount";
```

Comment:
Seems no one is using this config, are you going to achieve this logic in another PR?

Reply:
Yes, i will use them for #2385